### PR TITLE
Include active sanctions and QTS/EYTS in duplicate assessment task

### PR DESCRIPTION
**Pull request by gunndabad**
### Context

The task created to review a potential duplicate record should include whether the duplicate has active sanctions and/or QTS/EYTS date. 

https://trello.com/c/ttcI4bwG/180-fix-qts-task-copy-when-potential-duplicate-teacher-is-created-in-register-api

### Changes proposed in this pull request

Amends the description of the QTS task created to review a duplicate to include whether the duplicate has active sanctions and/or QTS/EYTS.

### Checklist

-   [x] Attach to Trello card
-   [x] Rebased master
-   [x] Cleaned commit history
-   [x] Tested by running locally

### CODE REVIEW

**Comment by gunndabad** [comment]
[Fix QTS task copy when potential duplicate teacher is created in Register API](https://trello.com/c/ttcI4bwG/180-fix-qts-task-copy-when-potential-duplicate-teacher-is-created-in-register-api)

**Comment by gunndabad** [comment]
[Fix QTS task copy when potential duplicate teacher is created in Register API](https://trello.com/c/ttcI4bwG/180-fix-qts-task-copy-when-potential-duplicate-teacher-is-created-in-register-api)

File: tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs [hunk 1]
```diff
@@ -105,8 +122,11 @@ public async Task Given_details_that_does_match_existing_record_does_not_allocat
             Assert.Equal("Notification for QTS Unit Team", crmTask.Subject);
             Assert.Equal(_crmClientFixture.Clock.UtcNow, crmTask.ScheduledEnd);
 
+            var expectedDescription = $"Potential duplicate\nMatched on\n\t- First name: '{firstName}'\n\t- Last name: '{lastName}'\n\t- Date of birth: '{birthDate:dd/MM/yyyy}'\n" +
```
Comment (defong) [inline tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs:125]: Wanna add middle name for completeness?

Will this fall foul of PII/GDPR?

Should it be the `id` instead?

File: tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs [hunk 2]
```diff
@@ -62,8 +62,18 @@ public async Task Given_details_that_do_not_match_existing_records_allocates_trn
...
 
-        [Fact]
-        public async Task Given_details_that_does_match_existing_record_does_not_allocate_trn_and_creates_QTS_task()
+        [Theory]
+        [InlineData(false, false, false, "")]
+        [InlineData(true, false, false, "Matched record has active sanctions\n")]
+        [InlineData(false, true, false, "Matched record has QTS date\n")]
+        [InlineData(false, false, true, "Matched record has EYTS date\n")]
+        [InlineData(true, true, false, "Matched record has active sanctions & QTS date\n")]
+        [InlineData(true, false, true, "Matched record has active sanctions & EYTS date\n")]
```
Comment (defong) [inline tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs:71]:
```
        [InlineData(true, true, true, "Matched record has active sanctions & QTS date & EYTS date\n")]
```

There seems to be this hidden option.

Comment (gunndabad) [reply tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs:71]: In reality that combination won't happen - it'll be at most one of EYTS or QTS.

File: tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs (hunk 1 above)
Comment (gunndabad) [reply tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs:125]: > Wanna add middle name for completeness?
Yep good spot.

> Will this fall foul of PII/GDPR?
> Should it be the id instead?
It's ok - the data is already in CRM at this point, this is just to make reviewing the duplicate easier for the end user. This copy is taken from existing file-based implementation.

File: tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs (hunk 2 above)
Comment (defong) [reply tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs:71]: Can we just add a note for the next person, as `code vs reality` has a discrepancy.

File: tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs (hunk 1 above)
Comment (defong) [reply tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs:125]: > existing file-based implementation

Seems reasonable, assuming external sentry/3rd party error tracking, has already been evaluated.

**Review by defong — APPROVED** [review]

File: tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs (hunk 2 above)
Comment (gunndabad) [reply tests/DqtApi.Tests/DataverseIntegration/CreateTeacherTests.cs:71]: I've amended the implementation to better convey that `HasEytsDate` and `HasQtsDate` are mutually exclusive.

**Pull request merged**